features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Applications/Details/ProductionKeys.js

```diff
@@ -84,97 +155,175 @@ class ProductionKeys extends React.Component {
-                                    value={consumerKey || "Keys are not generated yet. Click the Generate token button to generate the keys."}
-                                    className={"textField"}
-                                    helperText="Consumer Key of the application"
-                                    margin="normal"
-                                    fullWidth={true}
-                                />
+                            <TextField
+                                inputProps={{ readonly: true }}
+                                label="Consumer Key"
+                                id="consumerKey"
+                                value={consumerKey || "Keys are not generated yet. Click the Generate token button to generate the keys."}
+                                className={"textField"}
+                                helperText="Consumer Key of the application"
+                                margin="normal"
+                                fullWidth={true}
+                            />
                         </Grid>
                         <Grid item xs={7}>
-                                <FormControl fullWidth={true} margin="normal">
-                                    <InputLabel htmlFor="consumerSecret">Consumer Token</InputLabel>
-                                    <Input
-                                        inputProps={{readonly: true}}
-                                        id="consumerSecret"
-                                        type={(this.state.showCS || !consumerSecret) ? 'text' : 'password'}
-                                        value={consumerSecret || "Keys are not generated yet. Click the Generate token button to generate the keys."}
-                                        endAdornment={
-                                            <InputAdornment position="end">
-                                                <IconButton classes="" onClick={this.handleShowCS}
-                                                            onMouseDown={this.handleMouseDownGeneric}>
-                                                    {this.state.showCS ? <VisibilityOff/> : <Visibility/>}
-                                                </IconButton>
-                                            </InputAdornment>
-                                        }
-                                    />
-                                    <FormHelperText>Consumer Secret of the application</FormHelperText>
-                                </FormControl>
+                            <FormControl fullWidth={true} margin="normal">
+                                <InputLabel htmlFor="consumerSecret">Consumer Token</InputLabel>
+                                <Input
+                                    inputProps={{ readonly: true }}
+                                    id="consumerSecret"
+                                    type={(this.state.showCS || !consumerSecret) ? 'text' : 'password'}
```

Comment:
shall we get this to a variable and use it here?
@tmkasun wdyt?

Reply:
Yes +1 , It would be more readable then

```javascript
const showConsumerSecret =  showCS || !consumerSecret ;
// and in JSX
<Input
type={showConsumerSecret ? 'text' : 'password'}
/>
```